sdcm/stress_thread.py

```diff
@@ -160,6 +161,13 @@ def _run_stress(self, node, loader_idx, cpu_idx, keyspace_idx):
                                      loader_idx=loader_idx, cpu_idx=cpu_idx), \
                 CassandraStressEventsPublisher(node=node, cs_log_filename=log_file_name):
 
+            loader_pids = self.pids.get(loader_idx, None)
+            if loader_pids is None:
+                loader_pids = self.pids[loader_idx] = []
+            loader_pids.append(node.remoter.run(cmd = 'echo $$',
```

Comment:
Option duration= for cassandra-stress does not properly work in user profile case, instead of stoping process it just printing statistics and go another round.

Reply:
pass a timeout argument to the `run_stress_thread()` to limit it, and use the `verify_results` or `get_results` to wait for the c-s to finish.

AFAIK the c-s user profile duration param does work, this the template of command I've used before and it was working, and stopping according to duration 
```
cassandra-stress user profile={} cl=QUORUM 'ops(insert=1, read1=5)' duration={} -rate threads=2 -errors ignore
``` 
maybe this part in your command `no-warmup cl=ALL` is changing some behavior